### ISSUE TITLE
Claim depth malfunctioning - some aspects of claim always extend 100% vert regardless of extendintogrounddistance and/or maximumdepth 

### ISSUE THREAD

**Issue by niknagz** [GriefPrevention/GriefPrevention#1373]
### What happened:
Certain attributes of claims do not respect Y limits. Claim depth set to 5, but no blocks beneath it can be broken all the way down to bedrock. Inspecting blocks down to bedrock shows the claim, even when below the expected claim bottom. However, when trying to delete the claim while below expected claim bottom, it says there is no claim present, even though inspecting with a stick or building in the same place shows the claim. A similar issue I noticed (not currently bothering me) is that Max Claim Depth is not being respected in extremely similar way (claims can be made below the limit and inspected, but deleting them says no claim).

In sum: claims extend to bedrock regardless of config, though some aspects of claims (where you can delete from, etc.) respect depth limits.

### What was expected:
Users should be able to build when beneath a claim's lower y limit. Claim existence should not depend on whether user is inspecting or building vs deleting.

### Steps to reproduce:

- Brand new server, no other plugins, unedited GriefPrevention folder/config (extendintogrounddistance=5. 
- One player creates claim with golden shovel (e.g. y=70). 
- Other player tries to dig anywhere below claim (beyond expected Y limit of claim, e.g. y=10) and cannot dig. Receives message that the do not have permission to build from first player.
- Standing at same point (y=10) cannot delete claim, but can see it with STICK.

### Server and GriefPrevention version:
___This can be found by running `/version` and `/version GriefPrevention` on your server.___
Paste of `/version`: This server is running Paper version git-Paper-742 (MC: 1.16.5) (Implementing API version 1.16.5-R0.1-SNAPSHOT)
You are running the latest version
Paste of `/version GriefPrevention`: GriefPrevention version 16.17.1-2b69191

### Stack trace/error or server log

```
[18:41:52] [main/INFO]: Environment: authHost='https://authserver.mojang.com', accountsHost='https://api.mojang.com', sessionHost='https://sessionserver.mojang.com', servicesHost='https://api.minecraftservices.com', name='PROD'
[18:41:52] [main/INFO]: Found new data pack file/bukkit, loading it automatically
[18:41:52] [main/INFO]: Reloading ResourceManager: Default, bukkit
[18:41:53] [Worker-Main-5/INFO]: Loaded 7 recipes
[18:41:54] [Server thread/INFO]: Starting minecraft server version 1.16.5
[18:41:54] [Server thread/INFO]: Loading properties
[18:41:54] [Server thread/INFO]: This server is running Paper version git-Paper-742 (MC: 1.16.5) (Implementing API version 1.16.5-R0.1-SNAPSHOT)
[18:41:55] [Server thread/INFO]: Server Ping Player Sample Count: 12
[18:41:55] [Server thread/INFO]: Using 4 threads for Netty based IO
[18:41:55] [Server thread/INFO]: Debug logging is disabled
[18:41:55] [Server thread/INFO]: Default game type: SURVIVAL
[18:41:55] [Server thread/INFO]: Generating keypair
[18:41:55] [Server thread/INFO]: Starting Minecraft server on 51.161.84.225:25618
[18:41:55] [Server thread/INFO]: Using epoll channel type
[18:41:56] [Server thread/INFO]: [GriefPrevention] Loading GriefPrevention v16.17.1-2b69191
[18:41:56] [Server thread/INFO]: Preparing level "world"
[18:42:01] [Server thread/WARN]: Unable to find spawn biome
[18:42:02] [Server thread/WARN]: Unable to find spawn biome
[18:42:02] [Server thread/INFO]: Preparing start region for dimension minecraft:overworld
[18:42:03] [Server thread/INFO]: Loaded 0 spawn chunks for world world
[18:42:03] [Worker-Main-7/INFO]: Preparing spawn area: 0%
[18:42:03] [Server thread/INFO]: Time elapsed: 451 ms
[18:42:03] [Server thread/INFO]: Preparing start region for dimension minecraft:the_nether
[18:42:03] [Server thread/INFO]: Loaded 0 spawn chunks for world world_nether
[18:42:03] [Worker-Main-7/INFO]: Preparing spawn area: 0%
[18:42:03] [Worker-Main-1/INFO]: Preparing spawn area: 0%
[18:42:04] [Server thread/INFO]: Time elapsed: 730 ms
[18:42:04] [Server thread/INFO]: Preparing start region for dimension minecraft:the_end
[18:42:04] [Server thread/INFO]: Loaded 0 spawn chunks for world world_the_end
[18:42:04] [Worker-Main-5/INFO]: Preparing spawn area: 0%
[18:42:04] [Server thread/INFO]: Time elapsed: 497 ms
[18:42:04] [Server thread/INFO]: [GriefPrevention] Enabling GriefPrevention v16.17.1-2b69191
[18:42:04] [Server thread/INFO]: [GriefPrevention] Finished loading configuration.
[18:42:04] [Server thread/INFO]: [GriefPrevention] 0 total claims loaded.
[18:42:05] [Server thread/INFO]: [GriefPrevention] Customizable messages loaded.
[18:42:05] [Server thread/INFO]: [GriefPrevention] Finished loading data (File Mode).
[18:42:05] [Server thread/INFO]: [GriefPrevention] Boot finished.
[18:42:05] [Server thread/INFO]: Running delayed init tasks
[18:42:05] [Server thread/INFO]: Done (11.229s)! For help, type "help"
[18:42:05] [Server thread/WARN]: [io.papermc.paper.util.PaperJvmChecker] ************************************************************
[18:42:05] [Server thread/WARN]: [io.papermc.paper.util.PaperJvmChecker] * WARNING - YOU ARE RUNNING AN OUTDATED VERSION OF JAVA.
[18:42:05] [Server thread/WARN]: [io.papermc.paper.util.PaperJvmChecker] * PAPER WILL STOP BEING COMPATIBLE WITH THIS VERSION OF
[18:42:05] [Server thread/WARN]: [io.papermc.paper.util.PaperJvmChecker] * JAVA WHEN MINECRAFT 1.17 IS RELEASED.
[18:42:05] [Server thread/WARN]: [io.papermc.paper.util.PaperJvmChecker] *
[18:42:05] [Server thread/WARN]: [io.papermc.paper.util.PaperJvmChecker] * Please update the version of Java you use to run Paper
[18:42:05] [Server thread/WARN]: [io.papermc.paper.util.PaperJvmChecker] * to at least Java 16. When Paper for Minecraft 1.17 is
[18:42:05] [Server thread/WARN]: [io.papermc.paper.util.PaperJvmChecker] * released support for versions of Java before 16 will
[18:42:05] [Server thread/WARN]: [io.papermc.paper.util.PaperJvmChecker] * be dropped.
[18:42:05] [Server thread/WARN]: [io.papermc.paper.util.PaperJvmChecker] *
[18:42:05] [Server thread/WARN]: [io.papermc.paper.util.PaperJvmChecker] * Current Java version: 1.8.0_211
[18:42:05] [Server thread/WARN]: [io.papermc.paper.util.PaperJvmChecker] *
[18:42:05] [Server thread/WARN]: [io.papermc.paper.util.PaperJvmChecker] * Check this forum post for more information: 
[18:42:05] [Server thread/WARN]: [io.papermc.paper.util.PaperJvmChecker] *   https://papermc.io/java16
[18:42:05] [Server thread/WARN]: [io.papermc.paper.util.PaperJvmChecker] ************************************************************
[18:42:05] [Server thread/INFO]: Timings Reset


```
Or, paste the server.log at gist.github.com and paste the link here: 

### GriefPrevention config.yml

```yml

# Default values are perfect for most servers.  If you want to customize and have a question, look for the answer here first: http://dev.bukkit.org/bukkit-plugins/grief-prevention/pages/setup-and-configuration/
GriefPrevention:
  SeaLevelOverrides:
    world: -1
    world_nether: -1
    world_the_end: -1
  Claims:
    Mode:
      world_nether: Disabled
      world_the_end: Disabled
      world: Survival
    PreventGlobalMonsterEggs: true
    PreventTheft: true
    ProtectCreatures: true
    PreventButtonsSwitches: true
    LockWoodenDoors: false
    LockTrapDoors: false
    LockFenceGates: true
    EnderPearlsRequireAccessTrust: true
    RaidTriggersRequireBuildTrust: true
    ProtectHorses: true
    ProtectDonkeys: true
    ProtectLlamas: true
    InitialBlocks: 100
    Claim Blocks Accrued Per Hour:
      Default: 100
    Max Accrued Claim Blocks:
      Default: 80000
    Accrued Idle Threshold: 0
    AccruedIdlePercent: 0
    AbandonReturnRatio: 1.0
    AutomaticNewPlayerClaimsRadius: 4
    AutomaticNewPlayerClaimsRadiusMinimum: 0
    ExtendIntoGroundDistance: 5
    MinimumWidth: 5
    MinimumArea: 100
    MaximumDepth: 0
    InvestigationTool: STICK
    ModificationTool: GOLDEN_SHOVEL
    Expiration:
      ChestClaimDays: 7
      UnusedClaimDays: 14
      AllClaims:
        DaysInactive: 60
        ExceptWhenOwnerHasTotalClaimBlocks: 10000
        ExceptWhenOwnerHasBonusClaimBlocks: 5000
      AutomaticNatureRestoration:
        SurvivalWorlds: false
    AllowTrappedInAdminClaims: false
    MaximumNumberOfClaimsPerPlayer: 0
    CreationRequiresWorldGuardBuildPermission: true
    VillagerTradingRequiresPermission: true
    CommandsRequiringAccessTrust: /sethome
    DeliverManuals: true
    ManualDeliveryDelaySeconds: 30
    RavagersBreakBlocks: true
    FireSpreadsInClaims: false
    FireDamagesInClaims: false
    LecternReadingRequiresAccessTrust: true
  Spam:
    Enabled: true
    LoginCooldownSeconds: 60
    LoginLogoutNotificationsPerMinute: 5
    ChatSlashCommands: /me;/global;/local
    WhisperSlashCommands: /tell;/pm;/r;/whisper;/msg
    WarningMessage: Please reduce your noise level.  Spammers will be banned.
    BanOffenders: true
    BanMessage: Banned for spam.
    AllowedIpAddresses: 1.2.3.4; 5.6.7.8
    DeathMessageCooldownSeconds: 120
    Logout Message Delay In Seconds: 0
  PvP:
    RulesEnabledInWorld:
      world: true
      world_nether: true
      world_the_end: true
    ProtectFreshSpawns: true
    PunishLogout: true
    CombatTimeoutSeconds: 15
    AllowCombatItemDrop: false
    BlockedSlashCommands: /home;/vanish;/spawn;/tpa
    ProtectPlayersInLandClaims:
      PlayerOwnedClaims: true
      AdministrativeClaims: true
      AdministrativeSubdivisions: true
    AllowLavaDumpingNearOtherPlayers:
      PvPWorlds: true
      NonPvPWorlds: false
    AllowFlintAndSteelNearOtherPlayers:
      PvPWorlds: true
      NonPvPWorlds: false
    ProtectPetsOutsideLandClaims: false
  Economy:
    ClaimBlocksMaxBonus: 0
    ClaimBlocksPurchaseCost: 0.0
    ClaimBlocksSellValue: 0.0
  ProtectItemsDroppedOnDeath:
    PvPWorlds: false
    NonPvPWorlds: true
  BlockLandClaimExplosions: true
  BlockSurfaceCreeperExplosions: true
  BlockSurfaceOtherExplosions: true
  LimitSkyTrees: true
  LimitTreeGrowth: false
  PistonMovement: CLAIMS_ONLY
  PistonExplosionSound: true
  FireSpreads: false
  FireDestroys: false
  AdminsGetWhispers: true
  AdminsGetSignNotifications: true
  SmartBan: true
  Mute New Players Using Banned Words: true
  MaxPlayersPerIpAddress: 3
  Siege:
    Worlds: []
    BreakableBlocks:
    - GRASS_BLOCK
    - DIRT
    - COBBLESTONE
    - OAK_PLANKS
    - SPRUCE_PLANKS
    - BIRCH_PLANKS
    - JUNGLE_PLANKS
    - ACACIA_PLANKS
    - DARK_OAK_PLANKS
    - SAND
    - GRAVEL
    - GLASS
    - GRASS
    - FERN
    - DEAD_BUSH
    - WHITE_WOOL
    - ORANGE_WOOL
    - MAGENTA_WOOL
    - LIGHT_BLUE_WOOL
    - YELLOW_WOOL
    - LIME_WOOL
    - PINK_WOOL
    - GRAY_WOOL
    - LIGHT_GRAY_WOOL
    - CYAN_WOOL
    - PURPLE_WOOL
    - BLUE_WOOL
    - BROWN_WOOL
    - GREEN_WOOL
    - RED_WOOL
    - BLACK_WOOL
    - SNOW
    - GLASS_PANE
    DoorsOpenDelayInSeconds: 300
    CooldownEndInMinutes: 60
  EndermenMoveBlocks: false
  SilverfishBreakBlocks: false
  CreaturesTrampleCrops: false
  RabbitsEatCrops: true
  HardModeZombiesBreakDoors: false
  Database:
    URL: ''
    UserName: ''
    Password: ''
  UseBanCommand: false
  BanCommandPattern: ban %name% %reason%
  Advanced:
    fixNegativeClaimblockAmounts: true
    ClaimExpirationCheckRate: 60
    OfflinePlayer_cache_days: 90
  Abridged Logs:
    Days To Keep: 7
    Included Entry Types:
      Social Activity: true
      Suspicious Activity: true
      Administrative Activity: false
      Debug: false
      Muted Chat Messages: false


```

### Debugging steps attempted

- [ ] (If applicable) I attempted running the server without GriefPrevention installed
  - [ ] (If applicable) The problem does _not_ occur if GriefPrevention is removed from the server.
- [x] I attempted running only GriefPrevention on the server DONE
  - [x] Issue still occurs even if GriefPrevention is the only plugin running DONE
- [x] I attempted testing for the issue on a new server DONE
  - [x] Issue still occurs even after testing on a new server DONE
- [ ] I attempted testing for the issue with the vanilla client (no mods)
  - [ ] Issue still occurs even after testing with the vanilla Minecraft client

**Comment by niknagz** [comment]
Walking back GriefPrevention to the previous version fixes this. Newest version (16.17.1) has the prevailing issues mentioned above. Feel free to close or look into this accordingly

**Comment by Jikoo** [comment]
Please try a build including #1260.

**Comment by RoboMWM** [comment]
Assuming this is fixed with PR #1260 and possibly a duplicate of #1259 ?
